### DISCUSSION TITLE
Timetable for V1 Command Depreciation?

### DISCUSSION THREAD

**Discussion by liamzee** [haskell/cabal#9295] [Q&A] [answered]
When is V1 getting depreciated? If it gets depreciated, will it be removed from the codebase?

**Comment by Mikolaj** [comment, marked as answer]
Related: https://github.com/haskell/cabal/projects/12 and https://github.com/haskell/cabal/projects/10.

v1- is soft-deprecated already (I don't remember how --- removed from various forms of docs? any message on commandline?). It won't be removed until we have reasonable replacement for all functionality (but individual commands may be removed, e.g., if they don't work from some time already and nobody complains; there are some tickets about it). We don't for `cabal v1-install`, though `cabal-env` is close.
